### PR TITLE
Add validation of robot capabilities before starting mission

### LINKED ISSUE #134: Add validation of missions compatibility to robot
**Describe the new feature you would like to see**
ISAR should verify that a mission is compatible with the robot that ISAR is monitoring prior to the mission being sent to the state machine for execution. Validation implies checking that the robot implementation supports all the Tasks which the mission consists of.

**Describe the solution you'd like**
It is likely that this should be implemented with #95 such that ISAR may retrieve which Tasks the robot can perform.

**How will this feature affect the current Threat Model?**
No immediate effect on the current threat model.

### PR DESCRIPTION
**Pull request by vetlek**
Closes #134

### CODE REVIEW

File: src/isar/mission_planner/mission_validator.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import List
+
+from isar.models.mission import Mission
+
+
+def validate_robot_capabilities(
+    mission: Mission, robot_capabilities: List[str]
+) -> bool:
+    return all([task.type in robot_capabilities for task in mission.tasks])
```
Comment (tsundvoll) [inline src/isar/mission_planner/mission_validator.py:9]: Suggestion: Fetch the robot capabilities here instead
```suggestion
def is_robot_capable_of_mission(mission: Mission) -> bool:
    robot_capabilities = robot_settings.CAPABILITIES
    return all([task.type in robot_capabilities for task in mission.tasks])
```

**Comment by vetlek** [comment]
[Linked PR in isar-robot](https://github.com/equinor/isar-robot/pull/31)

**Comment by vetlek** [comment]
[Linked PR in isar-turtlebot](https://github.com/equinor/isar-turtlebot/pull/94)

File: src/isar/config/settings.py [hunk 2]
```diff
@@ -210,3 +210,28 @@ class Config:
...
+        super().__init__(_env_file=env_file_path)
+
+    CAPABILITIES: List[str] = Field(default_factory=default_capabilities)
+
+    class Config:
+        env_file_encoding = "utf-8"
+        case_sensitive = True
+
+
+robot_settings = RobotSettings()
```
Comment (tsundvoll) [inline src/isar/config/settings.py:237]: I suggest to make this a 'private' member function to keep it gathered together where it is used.
```suggestion

class RobotSettings(BaseSettings):
    def __init__(self) -> None:
        try:
            with pkg_resources.path(
                f"{settings.ROBOT_PACKAGE}.config", f"settings.env"
            ) as path:
                env_file_path = path
        except ModuleNotFoundError:
            env_file_path = None
        super().__init__(_env_file=env_file_path)

    def _default_capabilities():
        return ["drive_to_pose", "take_image"]

    CAPABILITIES: List[str] = Field(default_factory=_default_capabilities)

    class Config:
        env_file_encoding = "utf-8"
        case_sensitive = True


robot_settings = RobotSettings()
```

**Review by tsundvoll — APPROVED** [review]
I suggest to move the factory inside the class. When that is fixed, I approve of this PR.

File: src/isar/config/settings.py [hunk 3]
```diff
@@ -210,3 +210,24 @@ class Config:
...
+        try:
+            with pkg_resources.path(
+                f"{settings.ROBOT_PACKAGE}.config", f"settings.env"
+            ) as path:
+                env_file_path = path
+        except ModuleNotFoundError:
+            env_file_path = None
+        super().__init__(_env_file=env_file_path)
+
+    CAPABILITIES: List[str] = Field(default=["drive_to_pose", "take_image"])
```
Comment (tsundvoll) [inline src/isar/config/settings.py:226]: I would say this looks even better :)

**Pull request merged**